File: app-migration-sample/src/main/java/ru/surfstudio/android/app/migration/sample/ui/base/configurator/ActivityScreenConfigurator.java
package ru.surfstudio.android.app.migration.sample.ui.base.configurator;

import android.content.Intent;

import ru.surfstudio.android.app.migration.sample.app.App;
import ru.surfstudio.android.app.migration.sample.app.dagger.AppComponent;
import ru.surfstudio.android.app.migration.sample.ui.base.dagger.activity.ActivityComponent;
import ru.surfstudio.android.app.migration.sample.ui.base.dagger.activity.DaggerActivityComponent;
import ru.surfstudio.android.core.mvp.configurator.BaseActivityViewConfigurator;
import ru.surfstudio.android.sample.dagger.ui.base.dagger.activity.DefaultActivityModule;
import ru.surfstudio.android.sample.dagger.ui.base.dagger.screen.DefaultActivityScreenModule;

/**
 * Базовый конфигуратор для экрана, основанного на активити
 */

public abstract class ActivityScreenConfigurator
        extends BaseActivityViewConfigurator<AppComponent, ActivityComponent, DefaultActivityScreenModule> {

    public ActivityScreenConfigurator(Intent intent) {
        super(intent);
    }

    @Override
    protected ActivityComponent createActivityComponent(AppComponent parentComponent) {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .defaultActivityModule(new DefaultActivityModule(getPersistentScope()))
                .build();
    }

    @Override
    protected AppComponent getParentComponent() {
        return ((App) (getTargetActivity()).getApplicationContext()).getAppComponent();
    }

    @Override
    protected DefaultActivityScreenModule getActivityScreenModule() {
        return new DefaultActivityScreenModule(getPersistentScope());
    }
}

File: navigation/sample-standard/src/main/java/ru/surfstudio/android/navigation/sample_standard/dagger/ui/configurator/ActivityScreenConfigurator.kt
package ru.surfstudio.android.navigation.sample_standard.dagger.ui.configurator

import android.content.Intent
import ru.surfstudio.android.core.mvp.configurator.BaseActivityViewConfigurator
import ru.surfstudio.android.navigation.sample_standard.App
import ru.surfstudio.android.navigation.sample_standard.dagger.AppComponent
import ru.surfstudio.android.navigation.sample_standard.dagger.ui.ActivityComponent
import ru.surfstudio.android.navigation.sample_standard.dagger.ui.ActivityNavigationModule
import ru.surfstudio.android.navigation.sample_standard.dagger.ui.DaggerActivityComponent
import ru.surfstudio.android.sample.dagger.app.DefaultApp
import ru.surfstudio.android.sample.dagger.ui.base.dagger.activity.DefaultActivityModule
import ru.surfstudio.android.sample.dagger.ui.base.dagger.screen.DefaultActivityScreenModule

abstract class ActivityScreenConfigurator(intent: Intent?) : BaseActivityViewConfigurator<AppComponent, ActivityComponent, DefaultActivityScreenModule>(intent) {

    override fun createActivityComponent(parentComponent: AppComponent): ActivityComponent {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .defaultActivityModule(DefaultActivityModule(persistentScope))
                .activityNavigationModule(ActivityNavigationModule())
                .build()
    }

    override fun getParentComponent(): AppComponent {
        getTargetActivity()
        val app = persistentScope.screenState.activity.application as App
        return app.appComponent
    }

    override fun getActivityScreenModule(): DefaultActivityScreenModule {
        return DefaultActivityScreenModule(persistentScope)
    }
}